tests/e2e/specs/in-game/spectating.spec.js

```diff
@@ -360,3 +366,77 @@ describe('Spectating Games', () => {
     });
   });
 });
+
+describe('Creating And Updating Unranked Matches With Rematch - Spectating', () => {
+  beforeEach(function () {
+    cy.wipeDatabase();
+    cy.visit('/');
+
+    // Sign up players
+    cy.signupOpponent(playerOne).as('playerOneId');
+    cy.signupOpponent(playerThree).as('playerThreeId'); // spectator
+    // Opponent will be player 2 (the last one we log in as)
+    cy.signupOpponent(playerTwo).as('playerTwoId');
+
+    // Log in as playerOne
+    cy.loginPlayer(playerOne);
+    cy.setupGameAsSpectator();
+  });
+  it('Spectate unranked games with rematch', function () {
+    // 1st game: Opponent concedes
+    cy.recoverSessionOpponent(playerTwo);
+    cy.concedeOpponent();
+    assertVictory();
+    cy.log('rematch player2');
+
+    cy.rematchOpponent({ rematch: true });
+
+    cy.wait(1000);
```

Comment:
The purpose of this `cy.wait()` is to make sure that the `cy.rematchOpponent()` call is correctly processed before we move onto the `cy.window().its().then()` call starts making assertions on the DOM, basically to prevent the race condition of checking the store before the server has processed the opponent's rematch request. 


I think we can avoid the `cy.wait()` call by using an assertion on the DOM about what we expect to happen based on the above `cy.rematchOpponent()` call. That's considered the best practice because DOM assertions automatically retry until they succeed or timeout so that if it only takes say 300 milliseconds for the DOM assertion to pass we don't have to wait the full 1s before proceed.

In this case that should mean seeing the text that the opponent wants a rematch, right?
 so we could do this
```suggestion
    cy.get('[data-cy="opponent-wants-rematch"]').should('be.visible');
```

That way the assertion about the opponent wanting a rematch will retry until it passes (or times out after 4 seconds) so that the test can proceed earlier if the server takes less than 1s to respond. Check out the [cypress docs](https://docs.cypress.io/api/commands/wait#Time) on `cy.wait()` for more reading about the general idea

Reply:
This should apply to every use of `cy.wait()` that's been added I think, though the specific DOM assertion will depend on the step we are waiting for